src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -579,6 +579,18 @@ private function evaluate( $directive_value ) {
 		$path_segments = explode( '.', $path );
 		$current       = $store;
 		foreach ( $path_segments as $path_segment ) {
+			// Special case for numeric arrays to add a length property like JavaScript arrays.
+			if ( 'length' === $path_segment ) {
+				if ( is_array( $current ) && array_is_list( $current ) ) {
+					$current = count( $current );
+					break;
+				}
+				if ( is_string( $current ) ) {
+					$current = strlen( $current );
```

Comment:
Let's include a note in the code that explains that the result might differ linking to the Trac ticket or GitHub issue that covers the section starting with:

> Differences in string encodings may result in different results for the .length of a string between what the server reports and what JavaScript would detect.